src/lib/operators/case_operator.cpp

```diff
@@ -0,0 +1,226 @@
+#include "case_operator.hpp"
+
+#include <sstream>
+
+#include "storage/column_iterables/any_column_iterator.hpp"
+#include "storage/create_iterable_from_column.hpp"
+#include "storage/materialize.hpp"
+
+namespace {
+
+using namespace opossum;  // NOLINT
+
+/**
+ * Turns a PhysicalCaseExpressionResult - which is either a THEN result or an ELSE result - into a value/null - pair
+ */
+template <typename ResultDataType>
+std::pair<bool, ResultDataType> materialize_case_result(
+    const PhysicalCaseExpressionResult<ResultDataType>& case_result, const Chunk& chunk, const ChunkOffset chunk_offset,
```

Comment:
What is the difference between `CaseExpressionResult` and `PhysicalCaseExpressionResult`?